modules/mco-update-boot-images.adoc

```diff
@@ -0,0 +1,159 @@
+// Module included in the following assemblies:
+//
+// * post_installation_configuration/machine-configuration-tasks.adoc
+// * nodes/nodes-nodes-managing.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="mco-update-boot-images_{context}"]
+= Updating boot images
+
+A boot image is the image used by the Machine Config Operator (MCO) to boot up a {op-system-first} node. By default, the boot image is not managed by {product-title}. 
+
+This means that the boot image in your cluster is not updated along with your cluster. For example, if your cluster was originally created with {product-title} 4.11, the boot image used by the cluster is the same 4.11 version. When you scale up a new node, the MCO boots up that node with the originally-installed boot image and then automatically upgrades the node to the current cluster version. While this is a seamless process to the user, this process can cause issues such as the following conditions: 
+
+* Extra time to start up nodes
+* Certificate expiration issues
+* Version skew issues
+
+To avoid these issues, you can configure your cluster so that it updates the boot image anytime that you update your cluster. You can enable this feature by modifying the `MachineConfiguration` object. Currently, the ability to update the boot image is available for only Google Cloud Platform (GCP) clusters and is not supported for clusters managed by the Cluster CAPI Operator.
+
+:FeatureName: Boot image updating
+include::snippets/technology-preview.adoc[]
+
+You can view the current boot image used in your cluster by examining a machine set:
+
+.Example machine set with the boot image reference
+
+[source,yaml]
+----
+apiVersion: machine.openshift.io/v1beta1
+kind: MachineSet
+metadata:
+  name: ci-ln-hmy310k-72292-5f87z-worker-a
+  namespace: openshift-machine-api
+spec:
+# ...
+  template:
+# ...
+    spec:
+# ...
+      providerSpec:
+# ...
+        value:
+          disks:
+          - autoDelete: true
+            boot: true
+            projects/rhcos-cloud/global/images/rhcos-411-85-202203181601-0-gcp-x86-64 <1>
+# ...
+----
+<1> This boot image is the same as the installed {product-title} version, in this example, {product-title} 4.11. 
+
+If you configure your cluster to update your boot images, the boot image referenced in your machine sets is the same version as the cluster.
+
+If want to you opt out of this feature after some nodes have been updated with the new boot image, the nodes retain the updated boot image. There is no way to roll back the nodes to the original boot image version. Any new nodes created after opting out are created with the original boot image. Note that the boot image referenced in the machine set is not reverted, and continues to show the current cluster version.
```

Comment:
This is worded a little confusing. Here is an example that is hopefully helpful:

- MachineSet originally had boot image A. Any nodes scaled from this machineset will boot on image A.
- User upgrades the cluster version(not boot image) to image B.
- Any nodes scaled now will still be image A, as boot image updates are off and the machineset still references A.
- User now turns on boot image updates and opts in the MachineSet to boot image updates.
- MachineSet is now updated to boot image B. Any nodes scaled from this machineset in the future will have boot image B. As a result, the cluster will have a mixture of nodes, some born on A, some born on B.
- User now turns off boot image updates. The MachineSet will continue to reference bootimage B. Any nodes scaled up will still start from bootimage B, as the machineset refers to it.
- User upgrades the cluster to C. As the boot image update feature is off, the bootimage is still B, and any nodes launched from this machineset will still be born on B.

I hope that helps clarify things - the nodes will always start off based what is storied in the machineset. If the feature is turned off after some updates, any newly scaled nodes will start from the last update(B in this example) and not the latest. Turning off the feature will only prevent the machineset from being updated again - it does not mean that the machineset is restored back to the original image(A in the example).